utils/multimodal_dataset.py

```diff
@@ -0,0 +1,61 @@
+import torch
+from torch.utils.data import Dataset
+import pandas as pd
+import transformers
+import torchaudio
+from typing import Optional
+
+
+class TextAudioDataset(Dataset):
+    """Text and audio dataset for multimodal model, using BERT and wav2vec2."""
+
+    def __init__(self, df: pd.DataFrame, tokenizer, target_encoder, processor: transformers.Wav2Vec2Processor, max_len: Optional[int]):
+        """
+        Args:
+            :param df: dataframe of utterances and emotion label
+            :param tokenizer: tokenizer for creating embedding of utterances
+            :param target_encoder: encoding of text labels to integers e.g. instance of OneHotEncoder
+            :param processor: processor loaded from a pretrained model e.g.
+                              Wav2Vec2Processor.from_pretrained("facebook/wav2vec2-base-960h")
+            :param max_len: maximum length of embedding
+        """
+        self.df = df
+        self.text = df["text"].values

```

Comment:
it will be useful if we have a text light cleaning before we tokenize. Here is a link of a notebook that does the same https://github.com/Briiick/NLP-disaster-tweets/blob/main/notebooks/4-light-cleaning-BERT.ipynb